File: atividade_1/publisher.py
import paho.mqtt.client as mqtt
import time
import json


# Abre arquivo JSON
with open('sensor_data.json') as f:
    data = json.load(f)

sensor_data_list = data["sensor_data"]

# Configuração do cliente MQTT
client = mqtt.Client(mqtt.CallbackAPIVersion.VERSION2)

# Associa a função de callback ao cliente MQTT
# Conecta ao broker MQTT
client.connect("localhost", 1883, 60)

# Inicia o loop de comunicação MQTT
client.loop_start()

# Loop para publicar mensagens continuamente
try:
    while True:
        for sensor_data in sensor_data_list:
            message = " Valor: {} {}\n Timestamp: {}\n Localização: {}".format(sensor_data["value"], sensor_data["unit"], sensor_data["timestamp"], sensor_data["location"])
            client.publish("Bia", message)
            print(f"Leitura do sensor:\n\n{message}\n")
            time.sleep(2)
except KeyboardInterrupt:
    print("Publicação encerrada")

# Finaliza o loop de comunicação MQTT
client.loop_stop()

# Desconecta do broker MQTT
client.disconnect()

File: atividade_1/publisher.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"time"

	MQTT "github.com/eclipse/paho.mqtt.golang"
)

type SensorData struct {
	Value     float64 `json:"value"`
	Unit      string  `json:"unit"`
	Timestamp string  `json:"timestamp"`
	Location  string  `json:"location"`
}

func main() {
	// Abre arquivo JSON
	file, err := os.Open("sensor_data.json")
	if err != nil {
		panic(err)
	}
	defer file.Close()

	var sensorDataList []SensorData
	if err := json.NewDecoder(file).Decode(&sensorDataList); err != nil {
		panic(err)
	}

	// Configuração do cliente MQTT
	opts := MQTT.NewClientOptions().AddBroker("tcp://localhost:1883")
	opts.SetClientID("go_publisher")

	client := MQTT.NewClient(opts)
	if token := client.Connect(); token.Wait() && token.Error() != nil {
		panic(token.Error())
	}

	// Loop para publicar mensagens continuamente

		for {
			for _, sensorData := range sensorDataList {
				message := fmt.Sprintf("Valor: %.2f %s\nTimestamp: %s\nLocalização: %s", sensorData.Value, sensorData.Unit, sensorData.Timestamp, sensorData.Location)
				token := client.Publish("Bia", 0, false, message)
				token.Wait()
				fmt.Printf("Leitura do sensor:\n\n%s\n\n", message)
				time.Sleep(2 * time.Second)
			}
		}

		
			
	}
	
